cypress/e2e/pages/topicPage/tests.js

```diff
@@ -31,69 +31,73 @@ export default ({ service, pageType, variant }) => {
             otherVariant = variant === 'simp' ? 'trad' : 'simp';
           }
         }
-
         // Gets the topic page data for all the tests
-        cy.getPageData({
-          service,
-          pageType: 'topic',
-          variant,
-        }).then(({ body }) => {
-          topicTitle = body.data.title;
-          variantTopicId = body.data.variantTopicId;
-          pageCount = body.data.pageCount;
-          numberOfItems = body.data.curations[0].summaries.length;
-          firstItemHeadline = body.data.curations[0].summaries[0].title;
-          messageBanner = body.data.curations.find(
-            curation =>
-              curation.visualProminence === 'NORMAL' &&
-              curation.visualStyle === 'BANNER',
-          );
-        });
+        const data = await cy.getPageDataFromWindow();
+        const { pageData } = data;
+        topicTitle = pageData.title;
+        variantTopicId = pageData.scriptSwitchId;
+        pageCount = pageData.pageCount;
+        numberOfItems = pageData.curations?.[0]?.summaries.length;
+        firstItemHeadline = pageData.curations?.[0]?.summaries?.[0]?.title;
+        messageBanner = pageData.curations?.find(
+          curation =>
+            curation.visualProminence === 'NORMAL' &&
+            curation.visualStyle === 'BANNER',
+        );
+
         cy.log(`topic id ${topicId}`);
       }
     });
 
     describe(`Page content`, () => {
+      beforeEach(() => {
+        // make sure we always start from the path being tested to make the tests deterministic and not reliant on order
+        // as otherwise some tests can change the path and affect subsequent tests (i.e. when you change page script)
+        cy.visit(currentPath);
+      });
       it('should render a H1, which contains/displays topic title', () => {
         cy.log(Cypress.env('currentPath'));
-
         cy.get('h1').should('contain', topicTitle);
       });
+
       it('should render the correct number of items', () => {
         // Print SIMORGH_DATA if the number of promos on the page does not match
-        // the number of promos in the data from the BFF
+        // the number of promos in the data from window.SIMORGH_DATA
         // This is to help find out why sometimes a promo doesn't show on the page
-        cy.log(`Number of promos in BFF data${numberOfItems}`);
+        cy.log(`Number of promos in SIMORGH data${numberOfItems}`);
         const selector = '[data-testid="topic-promos"]:first > li';
         const promoCount = Cypress.$(selector).length;
         cy.log(`Number of promos on the page${promoCount}`);
-
         if (promoCount !== numberOfItems) {
           cy.window().then(win => {
             const pageData = win.SIMORGH_DATA;
```

Comment:
If we want to keep this log (I think we should, especially as we are seeing if this helps flakiness), shall we use the data we got on line 35?

Reply:
@LilyL0u I think we agreed that we no longer needed this log, because the number of items should always be consistent with the promoCount since the data is coming from window.SIMORGH_DATA and not potentially outdated / "too new" (i.e. data is newer than the page being tests) BFF data

But, we can leave it as is, and monitor it in Live for a while, and remove at a later date?